Core/src/org/sleuthkit/autopsy/sqlitereader/SQLiteReader.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2013-2018 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.sqlitereader;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FilenameUtils;
+import org.openide.util.NbBundle;
+
+/**
+ * Abstracts the need to open (and close) statements and result sets from the
+ * JDBC. Classes using the SQLiteExtractor must maintain a reference to the 
+ * Connection object generated in getDatabaseConnection.
+ */
+public final class SQLiteReader {
+    
+    private SQLiteReader(){}
+
+    public static Connection getDatabaseConnection(String dbPath) 
+            throws ClassNotFoundException, SQLException {
+        
+        // Load the SQLite JDBC driver, if necessary.
+        Class.forName("org.sqlite.JDBC"); //NON-NLS  
+        return DriverManager.getConnection(
+                "jdbc:sqlite:" + dbPath); //NON-NLS
+    }
```

Comment:
I think that we should relieve clients of the need to deal with the database connection. Instead, lets convert the class methods to instance methods, make the class implement AutoCloseable, and have the instances of the class hide and manage the connection on behalf of clients. I would add the writing of the database files to disk and the opening of the connection to the constructor. The user can supply the location for the write as a constructor argument.